### ISSUE TITLE
RDoc hangs (small reproducible sample)

### ISSUE THREAD

**Issue by jneen** [ruby/rdoc#410]
Users of rouge who don't use `--no-rdoc` currently can't install > 2.0 of the gem, because RDoc hangs trying to parse its source. I've reduced it to what I think is a minimal example - if I take out the `yield`, the `%<...>` string, or the interpolation, RDoc finishes and exits normally.

```
[ jneen@pepper testy ] $
; ls
doc/  testy.rb

[ jneen@pepper testy ] $
; cat testy.rb
class Test
  def thing
    yield %<<Hello, #{world}>>
  end
end

[ jneen@pepper testy ] $
; rdoc --version
4.2.2

[ jneen@pepper testy ] $
; rdoc .
Parsing sources...
^C0% [ 1/ 1]  testy.rb # <--- hangs, pegs cpu here
Interrupted

[ jneen@pepper testy ] $
;
```

**Comment by mailinglists35** [comment]
is this the root cause for https://github.com/rails/rails/issues/11814 ?

**Comment by aycabta** [comment]
OK, it's reproduced if RDoc 6.0.0.beta2. I'll check this bug later.

**Comment by aycabta** [comment]
I confused myself. This issue was still occurred with 5.1.0, but fixed with 6.0.0.beta2. So I'll close this issue, but I want to know that whether https://github.com/rails/rails/issues/11814 is fixed too or not.

**Comment by aycabta** [comment]
#526 and #528 fixes this issue.
